examples/workflow-glsp/src/workflow-views.tsx

```diff
@@ -34,24 +38,32 @@ import { ActivityNode, Icon, TaskNode, WeightedEdge } from './model';
 const JSX = { createElement: snabbdom.svg };
 
 @injectable()
-export class TaskNodeView extends RectangularNodeView {
-    render(node: TaskNode, context: RenderingContext): VNode {
-        const rcr = this.getRoundedCornerRadius(node);
+export class TaskNodeView extends RoundedCornerNodeView {
+    protected renderWithoutRadius(node: Readonly<SShapeElement & Hoverable & Selectable>, context: RenderingContext): VNode {
+        const task = node as TaskNode;
+        const rcr = this.getRoundedCornerRadius(task);
         const graph = <g>
-            <rect class-sprotty-node={true} class-task={true}
-                class-automated={node.taskType === 'automated'}
-                class-manual={node.taskType === 'manual'}
-                class-mouseover={node.hoverFeedback} class-selected={node.selected}
+            <rect class-sprotty-node={true}
                 x={0} y={0} rx={rcr} ry={rcr}
-                width={Math.max(0, node.bounds.width)} height={Math.max(0, node.bounds.height)}></rect>
+                {...this.additionalClasses(task, context)}
+                width={Math.max(0, node.bounds.width)} height={Math.max(0, node.bounds.height)} />
             {context.renderChildren(node)}
         </g>;
         return graph;
     }
 
-    protected getRoundedCornerRadius(node: SShapeElement): number {
+    protected getRoundedCornerRadius(_node: TaskNode): number {
         return 5;
     }
+
+    protected additionalClasses(element: Readonly<SShapeElement & Hoverable & Selectable>, context: RenderingContext): Classes {
+        const node = element as TaskNode;
+        return {
+            'class-task': true,
```

Comment:
Should be reworked with eclipse-glsp/glsp/issues/220 and applied with the GModelElement.cssClasses property